grafana-plugin/src/plugin/GrafanaPluginRootPage.tsx

```diff
@@ -45,54 +45,44 @@ export const GrafanaPluginRootPage = (props: AppRootProps) => (
   </Provider>
 );
 
+type PluginSetupWrapperProps = PropsWithChildren & {
+  text: string;
+};
+const PluginSetupWrapper: FC<PluginSetupWrapperProps> = ({ text, children }) => (
+  <div className="spin">
+    <img alt="Grafana OnCall Logo" src={logo} />
+    <div className="spin-text">{text}</div>
+    {children}
+  </div>
+);
+
 const RootWithLoader = observer((props: AppRootProps) => {
   const store = useStore();
 
+  const setupPlugin = useCallback(() => store.setupPlugin(props.meta), [props.meta]);
+
   useEffect(() => {
-    store.setupPlugin(props.meta);
+    setupPlugin();
   }, []);
 
   if (store.appLoading) {
-    let text = 'Initializing plugin...';
-
-    if (!store.pluginIsInitialized) {
-      text = '🚫 Plugin has not been initialized';
-    } else if (!store.correctProvisioningForInstallation) {
-      text = '🚫 Plugin could not be initialized due to provisioning error';
-    } else if (!store.correctRoleForInstallation) {
-      text = '🚫 Admin must sign on to setup OnCall before a Viewer can use it';
```

Comment:
all of the plugin initialization error messages have been moved to `store.initializationError`. If there was an error it is non-null, otherwise null. See `rootBaseStore.setupPlugin` for more details.